darts/models/forecasting_model.py

```diff
@@ -101,6 +105,343 @@ def _build_forecast_series(self,
+
+        # check forecast horizon‚
+        if hasattr(n, 'forecast_horizon'):
+            raise_if_not(n.forecast_horizon > 0, 'The provided forecasting horizon must be a positive integer.', logger)
+
+        # check start parameter
+        if hasattr(n, 'start'):
+            if isinstance(n.start, float):
+                raise_if_not(n.start >= 0.0 and n.start < 1.0, '`start` should be between 0.0 and 1.0.', logger)
+            elif isinstance(n.start, pd.Timestamp):
+                raise_if_not(n.start in training_series, '`start` timestamp is not in the `series`.', logger)
+                raise_if(n.start == training_series.end_time(), '`start` timestamp is the last timestamp of `series`',
+                         logger)
+            else:
+                raise_if(training_series[n.start].time_index()[0] == training_series.end_time(), '`start` timestamp '
+                         'is the last timestamp of `series`', logger)
+
+    def _backtest_model_specific_sanity_checks(self, *args: Any, **kwargs: Any) -> None:
+        """Method to be overriden in subclass for model specific sanity checks"""
+        pass
+
+    @_with_sanity_checks("_backtest_sanity_checks", "_backtest_model_specific_sanity_checks")
+    def backtest(self,
+                 training_series: TimeSeries,
+                 target_series: Optional[TimeSeries] = None,
+                 start: Union[pd.Timestamp, float, int] = 0.5,
+                 forecast_horizon: int = 1,
+                 stride: int = 1,
+                 retrain: bool = True,
+                 trim_to_series: bool = True,
+                 verbose: bool = False,
+                 use_full_output_length: Optional[bool] = None) -> Tuple[TimeSeries, TimeSeries]:
+        """ Retrain and forecast values pointwise with an expanding training window over `series`.
+
+        To this end, it repeatedly builds a training set from the beginning of `series`. It trains the current model on
+        the training set, emits a (point) prediction for a fixed forecast horizon, and then moves the end of the
+        training set forward by one time step. The resulting predictions are then returned.
+
+        Unless `retrain` is set to False, this always re-trains the models on the entire available history,
+        corresponding an expending window strategy.
+
+        If `retrain` is set to False (useful for models with many parameter such as `TorchForecastingModel` instances),
+        the model will only be trained only on the initial training window (up to `start` time stamp), and only if it
+        has not been trained before. Then, at every iteration, the newly expanded 'training sequence' will be fed to the
+        model to produce the new output.
+
+        Parameters
+        ----------
+        training_series
+            The training time series on which to backtest
+        target_series
+            The target time series on which to backtest
```

Comment:
Maybe mention for what kinds of models this one applies.

Reply:
> Maybe mention for what kinds of models this one applies.

Done. Could you check if the new parameter documentation for `target_series` looks good to you?

Reply:
lgtm, thanks!